### DISCUSSION TITLE
Create a matrix with a for loop

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#2975] [Q&A] [answered]
Consider the following code:

```
#let b = ((1,2), (3,4))

$ mat(
  #for c in b [
    $ W_(#c.at(0) #c.at(1)) "  " $ ;
]
```

My goal is to create a _vertical_ vector, by using the semi-colon. However, the semi-colon is printed out and no new row is added to the matrix:

![image](https://github.com/typst/typst/assets/75945439/128ac7dd-bcbb-4463-9e87-d68dbfc2c4e4)

Could you please help this `typst` newbie?

**Comment by xkevio** [comment, marked as answer]
This will transform `data` into an array of arrays, which can then be spread into the matrix where each array is its own line.

```typ
#let data = ((1,2), (3,4), (5,6))
#let m-data = data.map(x => ($W_(#x.at(0)#x.at(1))$,))
#let matrix = math.mat(delim: "[", ..m-data)

$ #matrix $
```

**Reply by psads-git** [reply]
Thanks a lot, Kevin!
